src/datasets/base.py

```diff
@@ -92,6 +139,45 @@ def get_vocab_sizes(self):
             .to_dict()
         )
 
+    def train_test_split(
+        self, test_size: float = 0.2, min_points: int = 10, min_trajectories: int = 10
+    ):
+        """Split the dataset into train and test sets, stratifying trajectories by label.
+
+        A stratified split on label to balance label classes across the
+        test and train sets. Random assignment is at the group (trajectory ID) level.
+
+        Parameters
+        ----------
+        test_size : float
+            The ratio of the data that should be assigned to the test set. Default is 20%.
+        min_points: int
+            The minimum number of points a single trajectory must have in order to be included.
+        min_trajectories : int
+            The minimum number of trajectories a subject must have in order to be included.
+        """
+        train_file = Path(f"data/{str(self)}_train.csv")
```

Comment:
Self would equal the "dataset_location" name? aka the object name as a string?

Ex: "geoLife_beijing"_test.csv or _train.csv